server/src/test/java/io/crate/execution/dml/IndexerTest.java

```diff
@@ -1208,44 +1177,6 @@ public void test_empty_arrays_together_with_another_field_added_as_new_cols_with
         );
     }
 
-    @Test
-    public void test_empty_arrays_are_prefixed_as_unknown() throws Exception {
-        SQLExecutor e = SQLExecutor.of(clusterService)
-                .addTable("create table tbl (i int) with (column_policy='dynamic')");
-        DocTableInfo table = e.resolveTableInfo("tbl");
-
-        Indexer indexer = new Indexer(
-            table.ident().indexNameOrAlias(),
-            table,
-            new CoordinatorTxnCtx(e.getSessionSettings()),
-            e.nodeCtx,
-            List.of(
-                table.getDynamic(ColumnIdent.of("empty_arr"), true, false)
-            ),
-            null
-        );
-        var item = item(List.of());
-        List<Reference> newColumns = indexer.collectSchemaUpdates(item);
-        assertThat(newColumns).hasSize(1);
-
-        ParsedDocument doc = indexer.index(item);
-        assertThat(doc.source().utf8ToString()).isEqualToIgnoringWhitespace(
-                """
-                {"_u_empty_arr":[]}
-                """
```

Comment:
Super minor: let's update javadoc of `DocTableInfo.lookupNameBySourceKey`
> *  <li>Empty arrays, or arrays with only null values</li> 

is not true anymore